homeassistant/components/plugwise/climate.py

```diff
@@ -48,94 +53,143 @@ def __init__(
+            hvac_modes.append(HVACMode.AUTO)
+
+        return hvac_modes
 
     @property
-    def preset_mode(self) -> str | None:
+    def preset_mode(self) -> str:
         """Return the current preset mode."""
-        return self.device.get("active_preset")
+        return self.device["active_preset"]
 
     @property
-    def extra_state_attributes(self) -> Mapping[str, Any] | None:
-        """Return entity specific state attributes."""
-        return {
-            "available_schemas": self.device.get("available_schedules"),
-            "selected_schema": self.device.get("selected_schedule"),
-        }
+    def supported_features(self) -> int:
+        """Return the supported features."""
+        features: int = ClimateEntityFeature.TARGET_TEMPERATURE
+        if self.hc_data.get("elga_cooling_enabled", False):
+            features = ClimateEntityFeature.TARGET_TEMPERATURE_RANGE
+        if self.device.get("preset_modes"):
+            features |= ClimateEntityFeature.PRESET_MODE
+
+        return features
+
+    @property
+    def target_temperature(self) -> float | None:
+        """Return the temperature we try to reach.
+
+        Connected to the HVACModes combinations of AUTO/HEAT and AUTO/COOL.
+        """
+
+        return self.device["thermostat"].get("setpoint")
+
+    @property
+    def target_temperature_high(self) -> float | None:
+        """Return the temperature we try to reach in case of cooling.
+
+        Connected to the HVACMode combination of AUTO/HEAT_COOL.
+        """
+        return self.device["thermostat"].get("setpoint_high")
+
+    @property
+    def target_temperature_low(self) -> float | None:
+        """Return the heating temperature we try to reach in case of heating.
+
+        Connected to the HVACMode combination AUTO/HEAT_COOL.
+        """
+        return self.device["thermostat"].get("setpoint_low")
 
     @plugwise_command
     async def async_set_temperature(self, **kwargs: Any) -> None:
         """Set new target temperature."""
-        if ((temperature := kwargs.get(ATTR_TEMPERATURE)) is None) or (
-            self._attr_max_temp < temperature < self._attr_min_temp
-        ):
-            raise ValueError("Invalid temperature requested")
-        await self.coordinator.api.set_temperature(self.device["location"], temperature)
+        if ATTR_HVAC_MODE in kwargs:
+            await self.async_set_hvac_mode(kwargs[ATTR_HVAC_MODE])
+
+        data: dict[str, Any] = {}
+        if ATTR_TEMPERATURE in kwargs:
+            data["setpoint"] = kwargs[ATTR_TEMPERATURE]
+        if ATTR_TARGET_TEMP_HIGH in kwargs:
+            data["setpoint_high"] = kwargs[ATTR_TARGET_TEMP_HIGH]
+        if ATTR_TARGET_TEMP_LOW in kwargs:
+            data["setpoint_low"] = kwargs[ATTR_TARGET_TEMP_LOW]
+
+        for _, temperature in data.items():
+            if not (self._attr_min_temp <= temperature <= self._attr_max_temp):
+                raise ValueError("Invalid temperature change requested")
```

Comment:
Maybe this message needs to be extended to raise which one went out of bounds? As there are now potentially multiple temperatures passed.